pkg/controllers/updaterun/execution.go

```diff
@@ -161,17 +164,26 @@ func (r *Reconciler) executeUpdatingStage(
 			markClusterUpdatingFailed(clusterStatus, updateRun.Generation, unexpectedErr.Error())
 			return 0, fmt.Errorf("%w: %s", errStagedUpdatedAborted, unexpectedErr.Error())
 		}
+
 		finished, updateErr := checkClusterUpdateResult(binding, clusterStatus, updatingStageStatus, updateRun)
 		if finished {
 			finishedClusterCount++
 			continue
+		} else {
+			// If cluster update has been running for more than 1 minute, mark the update run as stuck.
+			timeElapsed := time.Since(clusterStartedCond.LastTransitionTime.Time)
```

Comment:
checkClusterUpdateResult wait until all the resources become available which depends on the "timeToReady" setting by the user. Therefore, 'updateRunStuckThreshold' has to be the greater of 60 seconds or that number

Reply:
I realized one thing: "timeToReady" is a rolling update specific. It's under `RollingUpdateConfig` and cannot be set when rolling type is set to External. The wait is done in the rollingUpdate controller, not in the workApplier. For untrackable resources, workApplier simply sets it as Available with reason set to "Availability not trackable".

Reply:
good point, does it mean that we need to add this to the update run API too or create some wait between clusters when there are untrackable resources?

Reply:
i notice this problem when reviewing the code. Should we keep the default value as 1 min to keep it consistent with the crp config? later, we can support the customized value?

Reply:
Yea, looks like we need to do similar thing in updateRun. That will be a new feature.